### ISSUE TITLE
Increasing of loss on yolov3-tiny-xnor

### ISSUE THREAD

**Issue by zrobotparking** [AlexeyAB/darknet#870]
I followed your suggestion and try to train yolov3-tiny-xnor in classify 5 kinds of object.
Things work well if I only change few convolution layer into xnor.
But if I change all convolution layer, except first one, into xnor,the loss can't convergence .
Can you give me some advice?

![image](https://user-images.githubusercontent.com/11483741/40339998-734d751e-5daf-11e8-9520-e10faeff57e9.png)

Train on VOC dataset

**Comment by AlexeyAB** [comment]
Two convolutional layers should be **without** `xnor=1`, first and last.
So **1st** and **last** conv-layers should be Float-32 (32 bit), and others XNOR=1 (1-bit).

As here:
[tiny-yolo-voc_xnor.zip](https://github.com/AlexeyAB/darknet/files/2026487/tiny-yolo-voc_xnor.zip)

As you can see:
* **1st** layer takes only: `2.14%` of total computation cost
* **last** layer takes only: `0.62%` of total computation cost

So it doesn't decrease speed too much.

![image](https://user-images.githubusercontent.com/4096485/40361507-3c4633f8-5dd2-11e8-80b9-9495d4390541.png)

**Comment by zrobotparking** [comment]
@AlexeyAB 
I do keep 1st and last conv-layers in Float-32, 
this network is modify from yolov3-tiny
`[net]
# Training
batch=64
subdivisions=8
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.0005
burn_in=1000
max_batches = 500200
policy=steps
steps=400000,450000
scales=.1,.1

[convolutional]
batch_normalize=1
filters=16
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
xnor=1
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
xnor=1
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
xnor=1
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
xnor=1
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
xnor=1
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=1

[convolutional]
xnor=1
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

###########

[convolutional]
xnor=1
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
xnor=1
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
xnor=1
size=1
stride=1
pad=1
#filters=255
filters=30
activation=linear



[yolo]
mask = 3,4,5

anchors = 1.08,1.19, 3.42,4.41, 6.63,11.38, 9.42,5.11, 16.62,10.52
classes=5
#anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
#classes=80
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1

[route]
layers = -4

[convolutional]
xnor=1
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[upsample]
stride=2

[route]
layers = -1, 8

[convolutional]
xnor=1
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
#filters=255
filters=30

activation=linear

[yolo]
mask = 1,2,3
anchors = 1.08,1.19, 3.42,4.41, 6.63,11.38, 9.42,5.11, 16.62,10.52
classes=5
#anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
#classes=80

num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1
`
